src/System.Private.CoreLib/src/System/Runtime/CompilerServices/AsyncMethodBuilder.cs

```diff
@@ -1064,6 +1048,29 @@ internal static void ThrowAsync(Exception exception, SynchronizationContext targ
             }
         }
 
+        private static ExceptionDispatchInfo RunReturningAnyException<TStateMachine>(ref TStateMachine stateMachine) where TStateMachine : IAsyncStateMachine
```

Comment:
This extra generic method is questionable because of the extra runtime metadata that comes with every generic method.

I think there is a good idea here to just have a small try/catch instead of finally. What would the gains look like if the same small try/catch is inline in the caller?